Src/BootCamp.Chapter/CaesarCipher.cs

```diff
@@ -0,0 +1,102 @@
+﻿using System;
+using System.Text;
+
+namespace BootCamp.Chapter
+{
+    class CaesarCipher
+    {
+        public static void Demo()
+        {
+            bool isContinue = true;
+            while (isContinue)
+            {
+                int shift = UserSelectSizeAndDirectionOfShift();
+                string forOperation = RequestInput();
+                if (shift == 99)
+                {
+                    GenerateAllShiftIterations(forOperation);
+                }
+                else
+                {
+                    ConvertString(shift, forOperation);
+                }
+                isContinue = UserSelectsContinue();
+            }
+        }
+
+        private static void GenerateAllShiftIterations(string forOperation)
+        {
+            for (int i = 1; i <= 25; i++)
+            {
+                ConvertString(i, forOperation);
+            }
+        }
+
+        private static bool UserSelectsContinue()
+        {
+            Console.WriteLine("Please enter \"c\" to continue, any other key to exit");
+            string response = Console.ReadLine();
+            return ((response == "c") || (response == "C"));
+         }
+
+        private static int UserSelectSizeAndDirectionOfShift()
+        {
+            Console.WriteLine("Please indicate how many letters your cipher shifts (the key). " +
+                "This should be a range from -25 (the the left) to 25 (to the right) \r\n" +
+                "If you wish to decrypt a cipher with an unknown key, enter 99.");
+            
+            try
+            {
+                int shift = int.Parse(Console.ReadLine());
+                if (shift == 99) { return shift; }
+                if (Math.Abs(shift) > 25)
+                {
+                    Console.WriteLine($"Your cipher key, {shift} is greater than 25 and will cycle completely through the alphabet and result in a shift of {shift % 26}.");
+                    return (shift % 26);
+                }
+                return shift;
+            }
+            catch (Exception e)
```

Comment:
Handle exceptions for exceptional cases. Not having a number is not an exceptional case. If you can avoid basing logic in something other than an exception- you should. Catching exception is not expensive but throwing it is.

In this case, in particular, you should rather use .TryParse().